src/main/java/io/kontur/disasterninja/dto/layer/LayerSummaryDto.java

```diff
@@ -22,12 +30,82 @@ public class LayerSummaryDto {
     private final List<String> copyrights;
     private final boolean ownedByUser;
     private final ObjectNode featureProperties;
+    private final String mapboxStyle;
 
     public static LayerSummaryDto fromLayer(Layer layer) {
         return layer == null ? null : new LayerSummaryDto(layer.getId(), layer.getName(),
                 layer.getDescription(), layer.getCategory() == null ? null : layer.getCategory(),
                 layer.getGroup(), layer.isBoundaryRequiredForRetrieval(), layer.isEventIdRequiredForRetrieval(),
                 layer.getLegend(),
-                layer.getCopyrights(), layer.isOwnedByUser(), layer.getFeatureProperties());
+                layer.getCopyrights(), layer.isOwnedByUser(), layer.getFeatureProperties(), defineMapboxStyleForRequest(layer));
+    }
+
+    private static String defineMapboxStyleForRequest(Layer layer) {
+        if (layer.getMapboxStyles() == null || layer.getMapboxStyles().isNull() || layer.getMapboxStyles().isEmpty()) {
+            return null;
+        }
+
+        HttpServletRequest request = getCurrentHttpRequest();
+        if (request == null) {
+            return null;
+        }
+
+        String url = getMapboxUrl(layer);
+        if (isBlank(url)) {
+            return null;
+        }
+
+        List<String> supportedLanguages = getSupportedLanguages(layer);
+        if (CollectionUtils.isEmpty(supportedLanguages)) {
+            return url;
+        }
+
+        String userLanguage = request.getHeader("User-Language");
+        if (supportedLanguages.contains(userLanguage)) {
+            return url.replace("{lang}", userLanguage);
+        }
+
+        Enumeration<Locale> acceptLanguage = request.getLocales();
+        while (acceptLanguage.hasMoreElements()) {
+            String language = acceptLanguage.nextElement().getLanguage();
+            if (supportedLanguages.contains(language)) {
+                return url.replace("{lang}", language);
+            }
+        }
+
+        if (supportedLanguages.contains("en")) {
```

Comment:
is there a logic for non-translated basemap, where just name= tag is used and labels are different for each country?

Reply:
Nope. en is intended to be the last fallback style. And I didn't generate such style for basemap. Although there is some style_day.json that uses name for labels but it wasn't used for DN.